c/src/ml-api-service-training-offloading.c

```diff
@@ -0,0 +1,1058 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/**
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * @file ml-api-service-training-offloading.c
+ * @date 5 Apr 2024
+ * @brief ML training offloading service of NNStreamer/Service C-API
+ * @see https://github.com/nnstreamer/nnstreamer
+ * @author Hyunil Park <hyunil46.park@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <gio/gio.h>
+#include <gst/gst.h>
+#include <json-glib/json-glib.h>
+#include <nnstreamer-edge.h>
+
+#include "ml-api-internal.h"
+#include "ml-api-service.h"
+#include "ml-api-service-training-offloading.h"
+
+#define ML_SERVICE_OFFLOADING_TYPE_PIPELINE_RAW 3
+#define ML_SERVICE_OFFLOADING_TYPE_REPLY 5
+
+/** It(@~~@) will be replaced with the path set by the app.*/
+#define APP_RW_PATH "@APP_RW_PATH@"
+#define REMOTE_APP_RW_PATH "@REMOTE_APP_RW_PATH@"
+/** combined with trained model file name set in conf */
+#define TRAINED_MODEL_FILE "@TRAINED_MODEL_FILE@"
```

Comment:
Q1. If the user does not specify the path, where is the default path?
Q2. How can user set the each path? For example, `@APP_RW_PATH@` is set by `ml_service_training_offloading_set_path()`.

Reply:
User can set path by ```status = ml_service_set_information (sender_h, "path", file_path);```
and
``` status = ml_service_set_information (receiver_h, "path", file_path);```
the string ```@REMOTE_APP_RW_PATH@``` in the receiver pipe set by sender is changed to ```path```.
If the path is not set, a runtime error occurs.